server/src/main/java/org/elasticsearch/action/admin/cluster/health/ClusterHealthResponse.java

```diff
@@ -299,7 +304,14 @@ public String toString() {
 
     @Override
     public RestStatus status() {
-        return isTimedOut() ? RestStatus.REQUEST_TIMEOUT : RestStatus.OK;
+        if (RestApiVersion.current() == RestApiVersion.V_7 && isTimedOut()) {
+            deprecationLogger.critical(DeprecationCategory.API, "cluster_health_request_timeout",
+                "HTTP status code for an internal cluster health timeout will change from 408 to 200 in a future major version");
+        }
+        if (Boolean.parseBoolean(System.getProperty("es.cluster_health.request_timeout_408", "false"))) {
```

Comment:
Best not to parse the property each time, instead we should have a static constructor which reads it into a static field. Moreover we should reject any value except `true` in that static constructor.

Reply:
Good point! I've moved the property to a static final field. Is there any standard ES way to validate the property to make sure we reject any value except `true`?

Reply:
Yes, but I think you need a static constructor rather than doing it in a one-liner. See e.g. this:

https://github.com/elastic/elasticsearch/blob/4ef3a58a46cc51abe0fb213d2502f21ad143fc58/server/src/main/java/org/elasticsearch/transport/TransportService.java#L74-L84

Also an invalid value should fail the node, but I don't think we construct the `ClusterHealthResponse` class early enough in the node's lifecycle to do that. `TransportClusterHealthAction` is likely a better bet.

Reply:
Done! I've decided to extract to a method.